src/types/string.rs

```diff
@@ -71,24 +71,17 @@ impl PyString {
         match self.to_string() {
             Ok(s) => s,
             Err(_) => {
-                unsafe {
-                    let py_bytes = ffi::PyUnicode_AsEncodedString(
-                        self.as_ptr(),
-                        CStr::from_bytes_with_nul(b"utf-8\0").unwrap().as_ptr(),
-                        CStr::from_bytes_with_nul(b"surrogatepass\0")
-                            .unwrap()
-                            .as_ptr(),
-                    );
-                    // Since we have a valid PyString and replace any surrogates, assume success.
-                    debug_assert!(!py_bytes.is_null());
-                    // ensure DECREF will be called
-                    gil::register_pointer(NonNull::new(py_bytes).unwrap());
```

Comment:
Usage of `register_pointer` was broken since my change to pointer drop immediately with GIL, suprised this test wasn't failing (probably luck).

I have rewritten this to use usual owned references.

Reply:
:+1: